gen/generate.jl

```diff
@@ -1,66 +1,35 @@
-using Clang.cindex
-using Clang.wrap_c
-using Printf
+using Pkg
+using Pkg.Artifacts
+using Clang.Generators
+using Clang.Generators.JLLEnvs
+using LibCURL_jll
 
-# Set these to correspond to your local filesystem's curl and clang include paths
-const CURL_PATH = "/usr/local/opt/curl/include/curl"
-const CLANG_INCLUDES = [
-    "/usr/local/opt/llvm/include",
-]
+cd(@__DIR__)
 
-const SRC_DIR = abspath(@__DIR__, "..", "src")
+artifact_toml = joinpath(dirname(Pkg.pathof(LibCURL_jll)), "..", "StdlibArtifacts.toml")
+artifact_dir = Pkg.Artifacts.ensure_artifact_installed("LibCURL", artifact_toml)
 
-headers = map(x -> joinpath(CURL_PATH, x), ["curl.h", "easy.h", "multi.h"])
-context = wrap_c.init(;
-    headers = headers,
-    clang_args = String[],
-    common_file = joinpath(SRC_DIR, "lC_common_h.jl"),
-    clang_includes = CLANG_INCLUDES,
-    clang_diagnostics = true,
-    header_wrapped = (top_header, cursor_header) -> in(cursor_header, headers),
-    header_library = header -> "libcurl",
-    header_outputfile = header -> joinpath(SRC_DIR, "lC_") * replace(basename(header), "." => "_") * ".jl",
-)
-context.options.wrap_structs = true
+include_dir = joinpath(artifact_dir, "include") |> normpath
+curl_h = joinpath(include_dir, "curl", "curl.h")
+@assert isfile(curl_h)
 
-begin
-    context.headers = [joinpath(CURL_PATH, "curl.h")]
-    run(context)
-end
-
-function write_constants(filename::AbstractString, startswith_identifier::AbstractString, exports_file)
-    open(filename, "r") do file
-        lines = split(read(file, String), "\n")
+# mprintf_h = joinpath(include_dir, "curl", "mprintf.h")
+# stdcheaders_h = joinpath(include_dir, "curl", "stdcheaders.h")
 
-        for line in lines
-            if startswith(line, startswith_identifier)
-                @printf exports_file "export %s\n" split(line, r" |\(")[2]
-            end
-        end
-    end
-end
+for target in JLLEnvs.JLL_ENV_TRIPLES
+    target == "aarch64-apple-darwin20" && continue
```

Comment:
A warning would be nice. Also, what's stopping support for Apple Silicon? Ideally an issue would be referenced here

Reply:
The cross-platform support is based on the gcc shards in BB, but BB doesn't seem to support `Apple Silicon` for now. 

In fact, `JLLEnvs.JLL_ENV_TRIPLES` currently only mapped to the output of `BinaryBuilder.supported_platforms()` which doesn't include `aarch64-apple-darwin20`, so this line will never be hit. As you noticed, this line itself serves as a warning.

Reply:
OK, it turns out that BB does support Apple Silicon as an experimental platform. I just added the support for Apple Silicon.